app/main/updater.js

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import os from 'os';
+import { setInterval } from 'timers';
+import { app, autoUpdater, dialog } from 'electron';
+import { version as appVersion, name as appName } from '../../package.json';
+
+// Check once an hour
+const UPDATE_CHECK_TIME = 1000 * 60 * 60;
+
+// Exe name is set to the same as the package name
+app.setAppUserModelId(`com.squirrel.${appName}.${appName}`);
+
+try {
+  autoUpdater.setFeedURL(`https://tofino-update-server.herokuapp.com/update/${os.platform()}_${os.arch()}/${appVersion}`);
+} catch (e) {
+  // This happens when the application isn't code signed on OSX.
+}
+
+autoUpdater.on('update-downloaded', () => {
+  dialog.showMessageBox({
+    type: 'question',
```

Comment:
It seems OK to have some UI that the normal frontend doesn't know about any actions taken in it will never affect the renderer, and is contained within a UI which is only ever rendered by main process.  If we end up wanting to show a UI for this in a content page then that's another story, of course.  

It's just going to be hard to test auto updating in general, especially if it involved quitting and restarting the process.